lib/dat-manager.js

```diff
@@ -44,102 +47,102 @@ module.exports = class Manager extends EventEmitter {
-      dat.db.close(err => {
-        if (err) return cb(err)
 
-        this.dats.splice(this.dats.indexOf(dat), 1)
-        dat.removeAllListeners('update')
+      this.dats.splice(this.dats.indexOf(dat), 1)
+      dat.events.removeAllListeners('update')
 
-        db.del(['archive', dat.key], (err) => {
-          if (err) return cb(err)
+      db.del(['archive', dat.key], (err) => {
+        if (err) return cb(err)
 
-          this.emit('update')
-          cb()
-        })
+        this.emit('update')
+        cb()
       })
     })
   }
-  _load ({ key, owner, dir }) {
+  _load ({ key, dir }, cb) {
     if (Buffer.isBuffer(key)) key = encoding.encode(key)
-    const dat = Dat({ key, dir })
-    dat.dir = dir
-    dat.owner = owner
-    if (dat.owner) {
-      dat.share(err => {
-        if (err) dat.emit('error', err)
-      })
-    } else {
-      dat.download(err => {
-        if (err) dat.emit('error', err)
-      })
-    }
-    return dat
+    Dat(dir, { key }, (err, dat) => {
+      if (err) return cb(err)
+
+      dat.dir = dir
+      dat.joinNetwork()
+
+      if (dat.owner) {
+        dat.importFiles(err => {
+          if (err) this.emit('error', err)
+        })
+      }
+      cb(null, dat)
+    })
   }
   _loadAll (cb) {
     const dats = []
     db.createReadStream({
       gt: ['archive', null],
       lt: ['archive', undefined]
     })
-    .on('data', data => {
-      dats.push(this._activate(this._load(data.value)))
-    })
+    .pipe(Transform({
+      objectMode: true,
+      transform: (data, _, done) => {
+        this._load(data.value, (err, dat) => {
+          if (err) return cb(err)
+          dats.push(this._activate(dat))
+          done()
+        })
+      }
+    }))
     .on('error', err => cb(err))
-    .on('end', () => cb(null, dats))
+    .on('finish', () => cb(null, dats))
   }
-  _persist ({ key, owner, dir }, cb) {
+  _persist ({ key, dir }, cb) {
     db.put(['archive', key], {
       key: encoding.encode(key),
-      owner,
       dir
     }, cb)
   }
   _activate (dat) {
-    dat.open(err => {
-      if (err) throw err
-      dat.listStream = dat.archive.list({ live: true })
-      dat.listStream.on('data', entry => {
-        if (entry.name !== 'dat.json') return
-        collect(dat.archive.createFileReadStream('dat.json'), (err, raw) => {
-          if (err) return console.error('failed to read dat.json', err)
-          let json
-          try {
-            json = JSON.parse(raw.toString())
-          } catch (err) {
-            console.error('failed to parse dat.json', err)
-            return
-          }
-          dat.title = json.title
-          dat.author = json.author
-          dat.emit('update')
-        })
+    dat.events = new EventEmitter()
+    dat.listStream = dat.archive.list({ live: true })
+    dat.listStream.on('data', entry => {
+      if (entry.name !== 'dat.json') return
+      collect(dat.archive.createFileReadStream('dat.json'), (err, raw) => {
```

Comment:
Ooh nice. So if there is an update to `dat.json` (i.e. more than one `dat.json` entry in metadata), the UI will just keep updating until the last metadata entry?